examples_tests/13.BRDF_Explorer/CBRDFBuiltinIncludeLoader.h

```diff
@@ -61,26 +69,101 @@ float _GGXSmith_G1_(in float a2, in float NdotX)
+}
 
 float GGXSmith(in float a2, in float NdotL, in float NdotV)
 {
     return _GGXSmith_G1_(a2, NdotL) * _GGXSmith_G1_(a2, NdotV);
 }
+float GGXSmith_wo_numerator(in float a2, in float NdotL, in float NdotV)
+{
+    return _GGXSmith_G1_wo_numerator(a2, NdotL) * _GGXSmith_G1_wo_numerator(a2, NdotV);
+}
+
+float GGXSmithHeightCorrelated(in float a2, in float NdotL, in float NdotV)
+{
+    float denom = NdotV*sqrt(a2 + (1.0 - a2)*NdotL*NdotL) + NdotL*sqrt(a2 + (1.0 - a2)*NdotV*NdotV);
+    return 2.0*NdotL*NdotV / denom;
+}
+
+float GGXSmithHeightCorrelated_wo_numerator(in float a2, in float NdotL, in float NdotV)
+{
+    float denom = NdotV*sqrt(a2 + (1.0 - a2)*NdotL*NdotL) + NdotL*sqrt(a2 + (1.0 - a2)*NdotV*NdotV);
+    return 1.0 / denom;
+}
+
+// Note a, not a2!
+float GGXSmithHeightCorrelated_approx(in float a, in float NdotL, in float NdotV)
+{
+    float num = 2.0*NdotL*NdotV;
+    return num / mix(num, NdotL+NdotV, a);
+}
+
+// Note a, not a2!
+float GGXSmithHeightCorrelated_approx_wo_numerator(in float a, in float NdotL, in float NdotV)
+{
+    return 1.0 / mix(2.0*NdotL*NdotV, NdotL+NdotV, a);
+}
+
+float GGXSmithHeightCorrelated_aniso_wo_numerator(in float at, in float ab, in float TdotL, in float TdotV, in float BdotL, in float BdotV, in float NdotL, in float NdotV)
+{
+    float Vterm = NdotL * length(vec3(at*TdotV, ab*BdotV, NdotV));
+    float Lterm = NdotV * length(vec3(at*TdotL, ab*BdotL, NdotL));
+    return 1.0 / (Vterm + Lterm);
+}
 
 #endif
 )";
     }
-    static std::string getFresnelSchlick(const std::string&)
+    static std::string getFresnel(const std::string&)
     {
         return
-R"(#ifndef _BRDF_SPECULAR_FRESNEL_FRESNEL_SCHLICK_INCLUDED_
-#define _BRDF_SPECULAR_FRESNEL_FRESNEL_SCHLICK_INCLUDED_
+R"(#ifndef _BRDF_SPECULAR_FRESNEL_FRESNEL_INCLUDED_
+#define _BRDF_SPECULAR_FRESNEL_FRESNEL_INCLUDED_
 
-vec3 FresnelSchlick(in vec3 F0, in float NdotV)
+vec3 FresnelSchlick(in vec3 F0, in float VdotH)
 {
-    float x = 1.0 - NdotV;
+    float x = 1.0 - VdotH;
     return F0 + (1.0 - F0) * x*x*x*x*x;
-} 
+}
+
+// code from https://seblagarde.wordpress.com/2013/04/29/memo-on-fresnel-equations/
+vec3 Fresnel_conductor(vec3 Eta, vec3 Etak, float CosTheta)
+{  
+   float CosTheta2 = CosTheta * CosTheta;
+   float SinTheta2 = 1.0 - CosTheta2;
+   vec3 Eta2 = Eta * Eta;
+   vec3 Etak2 = Etak * Etak;
+
+   vec3 t0 = Eta2 - Etak2 - SinTheta2;
+   vec3 a2plusb2 = sqrt(t0 * t0 + 4 * Eta2 * Etak2);
+   vec3 t1 = a2plusb2 + CosTheta2;
+   vec3 a = sqrt(0.5 * (a2plusb2 + t0));
+   vec3 t2 = 2 * a * CosTheta;
+   vec3 Rs = (t1 - t2) / (t1 + t2);
+
+   vec3 t3 = CosTheta2 * a2plusb2 + SinTheta2 * SinTheta2;
+   vec3 t4 = t2 * SinTheta2;   
+   vec3 Rp = Rs * (t3 - t4) / (t3 + t4);
+
+   return 0.5 * (Rp + Rs);
+}
+float Fresnel_dielectric(in float Eta, in float CosTheta)
+{
+   float SinTheta2 = 1.0 - CosTheta * CosTheta;
+
+   float t0 = sqrt(1.0 - (SinTheta2 / (Eta * Eta)));
+   float t1 = Eta * t0;
```

Comment:
We need Fresnel Diffuse Multiscatter Constant Polynomial Approximation in terms of `t1`